src/monitor.py

```diff
@@ -0,0 +1,271 @@
+        sys.exit(1)
+
+    # print time info for exerciser run being analyzed, and the current time
+    run_time = datetime.strptime(timestamp_dir.name, "%Y-%m-%d_%H-%M")
+    curr_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    print(f"Evaluating run from: {run_time}")
+    print(f"Current time is: {curr_time}")
+
+    # 2 dicts to store info on the status of the exerciser run
+    # 1 for expected tests (appears in working dir)
+    # 1 for unkown tests (doesn't appear in working dir, but does appear in shared log)
+    # both are dicts of dicts. each subdict stores info for a single test
+    # subdict fields are: submitted_resources, executed_resources,
+    # succeeded_resources, failed_resources, aborted_resources
+    expected_tests = {}
+    unknown_tests = {}
+    for item in timestamp_dir.iterdir():
+        # every dir represents a test
+        if item.is_dir():
+            test_dict = {
+                "submitted_resources": [],
+                "executed_resources": [],
+                "succeeded_resources": [],
+                "failed_resources": [],
+                "aborted_resources": [],
+            }
+            expected_tests[item.name] = test_dict
+
+    # clusters dict to store mapping of event cluster to test and associated procs
+    clusters = {}
+    event_log = JobEventLog(shared_log)
+    # loop through all events in shared event log, and filter for submit, execute, termination,
+    # and abortion events
+    for event in event_log.events(0):
+        # submit event: add test info to related dicts
+        if event.type is JobEventType.SUBMIT:
+            log_notes = event["LogNotes"]
+            colon_pos = log_notes.find(":")
+            if colon_pos != -1:
+                testname, resource = log_notes[colon_pos + 1 :].split(",")
+                if testname in expected_tests.keys():
+                    expected_tests[testname]["submitted_resources"].append(resource)
+                    # add info to clusters to utilize for future execute, term, and abort events
+                    if event.cluster not in clusters:
+                        clusters[event.cluster] = {
+                            "testname": testname,
+                            "known": True,
+                            "procs": {event.proc: resource},
+                        }
+                    else:
+                        clusters[event.cluster]["procs"][event.proc] = resource
+                else:
+                    unknown_tests[testname]["submitted_resources"].append(resource)
+                    if event.cluster not in clusters:
+                        clusters[event.cluster] = {
+                            "testname": testname,
+                            "known": False,
+                            "procs": {event.proc: resource},
+                        }
+                    else:
+                        clusters[event.cluster]["procs"][event.proc] = resource
```

Comment:
The cluster/proc storage is basically identical between the submit event expected or not. We should do this before the test name conditional defaulting the `"known"` value to `True` and setting to false in the unexpected test case.